app/helpers/application_helper/button/smart_state_scan.rb

```diff
@@ -0,0 +1,19 @@
+class ApplicationHelper::Button::SmartStateScan < ApplicationHelper::Button::Basic
+  needs :@record
+
+  SMART_ROLES = %w(SmartProxy SmartState).freeze
```

Comment:
@enoodle who are you asking? I assume you asked @roliveri (although I am sure you're more than qualified to look at the code and find a relevant place on your own).